### DISCUSSION TITLE
Setting different values ​​for min_scene_len results in the same content being divided into different scenes

### DISCUSSION THREAD

**Discussion by iaaidev1** [Breakthrough/PySceneDetect#469] [Q&A] [answered]
Using the default value of min_scene_len, the following results are obtained:

```
[
    ...
    {
        "url": "",
        "scene_detect": {
            "start_frame": 166,
            "start_time": "00:00:08.300",
            "end_frame": 245,
            "end_time": "00:00:12.250"
        }
    },
    {
        "url": "",
        "scene_detect": {
            "start_frame": 246,
            "start_time": "00:00:12.300",
            "end_frame": 295,
            "end_time": "00:00:14.750"
        }
    },
    ...
]
[240, 245] interval split in the previous interval.
```

min_scene_len is set to 2s (2 * video.frame_rate), and the following results are obtained:
```
[
    ...
    {
        "url": "",
        "scene_detect": {
            "start_frame": 185,
            "start_time": "00:00:09.250",
            "end_frame": 239,
            "end_time": "00:00:11.950"
        }
    },
    {
        "url": "",
        "scene_detect": {
            "start_frame": 240,
            "start_time": "00:00:12.000",
            "end_frame": 661,
            "end_time": "00:00:33.050"
        }
    },
    ...
]
[240, 245] interval split in the next interval。
```
video.frame_rate is 20.0

**Comment by Breakthrough** [comment]
What detector are you using?  There is a known issue for AdaptiveDetector in #408.

**Reply by iaaidev1** [reply]
> What detector are you using? There is a known issue for AdaptiveDetector in #408.
this is the code:

```python
content_detector = ContentDetector(threshold=threshold, min_scene_len=min_scene_len)
source_scene_list = detect(
    video_path=video_path,
    detector=content_detector,
    start_time=start_time,
    end_time=end_time,
    show_progress=True)
```

The problem can be reproduced on versions 0.6.4 and 0.6.5

**Reply by Breakthrough** [reply, marked as answer]
Any time the `min_scene_len` parameter is enforced, the results will differ since PySceneDetect needs to determine what to do.  In previous versions (v0.6.3 and older), any cut points that occurred before `min_scene_len` were ignored.

Newer versions (v0.6.4+) allow you to customize what happens with the `filter_mode` param.  By default, `ContentDetector` now *merges* cuts shorter than `min_scene_len` with the preceding scene.  This greatly improves performance on [videos with lots of flashing](https://github.com/Breakthrough/PySceneDetect/issues/35#issuecomment-2067517852).

You can still get the old behaviour by setting the `filter_mode` param of `ContentDetector` to `FlashFilter.Mode.SUPPRESS`, e.g.:

```python
from scenedetect.scene_detector import FlashFilter
content_detector = ContentDetector(..., filter_mode=FlashFilter.Mode.SUPPRESS)
```

To explain what both modes do:

- `FlashFilter.Mode.MERGE`: New default for 0.6.4+, *merges* scenes shorter than `min_scene_len` with previous scene
- `FlashFilter.Mode.SUPPRESS`: Previous default before 0.6.4, *ignores* new scenes until `min_scene_len` has passed

Let me know if this explains the differences you're seeing!

**Comment by iaaidev1** [comment]
I understand the problem, thank you very much.
